### PR TITLE
Use lru to cache blspubkey

### PR DESCRIPTION
**Pull request by LeoHChen**
Inline validator is used to solve the missing signature issue of leader.

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -650,12 +652,10 @@ func (node *Node) Start() error {
...
 
 		go func() {
-
-			for msg := range msgChan {
-				// should not take more than 3 seconds to process one message
-				ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
-				msg := msg
-
+			// should not take more than 5 seconds to process one message
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (rlan35) [inline node/node.go:656]: Does it work to have a single context object to use for all messages?

Comment (LeoHChen) [reply node/node.go]: fixed.

**Comment by LeoHChen** [comment]
image

File: crypto/bls/bls.go [hunk 2]
```diff
@@ -18,14 +26,28 @@ func RandPrivateKey() *bls.SecretKey {
...
 	}
 	pubKey := &bls.PublicKey{}
 	err := pubKey.Deserialize(bytes)
-	return pubKey, err
+
+	if err != nil {
+		return nil, err
+	}
+
+	BLSPubKeyCache.Add(kkey, pubKey)
```
Comment (JackyWYX) [inline crypto/bls/bls.go:49]: Can we use value instead of pointer for value caching?

Comment (LeoHChen) [reply crypto/bls/bls.go]: done

File: crypto/bls/bls.go [hunk 3]
```diff
@@ -18,14 +26,33 @@ func RandPrivateKey() *bls.SecretKey {
...
+			return &pk, nil
+		} else {
+			return nil, errPubKeyCast
+		}
 	}
 	pubKey := &bls.PublicKey{}
 	err := pubKey.Deserialize(bytes)
-	return pubKey, err
+
+	if err != nil {
```
Comment (JackyWYX) [inline crypto/bls/bls.go:50]: This is also open to attack. Can we also cache the error?

Comment (JackyWYX) [reply crypto/bls/bls.go:50]: Ummm, this seems cannot be fully mitigated until we implement the blacklist feature. So seems good now.

**Pull request merged**